src/compas_view2/scene/camera.py

```diff
@@ -123,42 +165,10 @@ def pan(self, dx, dy):
         None
 
         """
-        if self.view.current == self.view.PERSPECTIVE:
-            sinrz = sin(radians(self.rz))
-            cosrz = cos(radians(self.rz))
-            sinrx = sin(radians(self.rx))
-            cosrx = cos(radians(self.rx))
-            _dx = dx * cosrz - dy * sinrz * cosrx
-            _dy = dy * cosrz * cosrx + dx * sinrz
-            _dz = dy * sinrx * self.pan_delta
-            _dx *= 0.1 * self.distance
-            _dy *= 0.1 * self.distance
-            _dz *= 0.1 * self.distance
-            self.tx += self.pan_delta * _dx
-            self.ty -= self.pan_delta * _dy
-            self.target[0] = -self.tx
-            self.target[1] = -self.ty
-            self.target[2] -= _dz
-            self.distance -= _dz
-
-        elif self.view.current == self.view.FRONT:
-            _dx = dx * 0.1 * self.distance
-            _dz = dy * 0.1 * self.distance
-            self.tx += self.pan_delta * _dx
-            self.target[2] += 0.01 * _dz
-
-        elif self.view.current == self.view.RIGHT:
-            _dx = dx * 0.1 * self.distance
-            _dz = dy * 0.1 * self.distance
-            self.tx += self.pan_delta * _dx
-            self.target[2] += 0.01 * _dz
-
-        elif self.view.current == self.view.TOP:
-            self.tx += dx * self.pan_delta * 0.1 * self.distance
-            self.ty -= dy * self.pan_delta * 0.1 * self.distance
-
-        else:
-            raise NotImplementedError
+        R = Rotation.from_euler_angles(self.rotation)
+        T = Translation.from_vector([-dx * self.pan_delta, dy * self.pan_delta, 0])
+        _, _, _, translation, _ = decompose_matrix((R * T).matrix)
```

Comment:
decomposing a matrix is an expensive operation.
if only the translation is need perhaps just do
```python
M = (R * T).matrix
vector = [M[i][3] for i in range(3)]
```
or something like that...